library/general/src/lib/cfa/sysctl_config.rb

```diff
@@ -0,0 +1,209 @@
+  # other files.
+  #
+  # @example Enabling IPv4 forwarding
+  #   sysctl = SysctlConfig.new
+  #   sysctl.forward_ipv4 = true
+  #   sysctl.save
+  #
+  # Although in the previous example we enabled the IPv4 forwarding using by
+  # setting +forward_ipv4+ to true. However, under the hood, the kernel maps
+  # boolean values to "1" or "0". If you want to access to that raw value,
+  # you can prepend "raw_" to the method's name.
+  #
+  # @example Accessing the raw value of a setting
+  #   sysctl = SysctlConfig.new
+  #   sysctl.load
+  #   sysctl.raw_forward_ipv6 #=> "0"
+  #   sysctl.raw_forward_ipv6 = "1"
+  class SysctlConfig
+    include Yast::Logger
+    include Yast::I18n
+    extend Yast::I18n
+
+    PATHS = [
+      "/run/sysctl.d",
+      "/etc/sysctl.d",
+      "/usr/local/lib/sysctl.d",
+      "/usr/lib/sysctl.d",
+      "/lib/sysctl.d",
+      "/etc/sysctl.conf"
+    ].freeze
+    private_constant :PATHS
+
+    YAST_CONFIG_PATH = Sysctl::PATH
+    private_constant :YAST_CONFIG_PATH
+
+    class << self
+      def define_attr(attr)
+        define_method attr do
+          file = files.reverse.find do |f|
+            f.present?(attr)
+          end
+          return file.public_send(attr) if file
+
+          yast_config_file.public_send(attr)
+        end
+
+        define_method "#{attr}=" do |value|
+          yast_config_file.public_send("#{attr}=", value)
+        end
+      end
+    end
+
+    Sysctl.known_attributes.each { |a| define_attr(a) }
+
+    def load
+      files.each(&:load)
+    end
+
+    # Saving all sysctl settings
+    #
+    def save
+      yast_config_file&.save
+    end
+
+    # Whether there is a conflict with given attributes
+    #
+    # @param only [Array<Symbol>] attributes to check
+    # @param show_information [Boolean] showing a popup if it is conflicting
+    # @return [Boolean] true if any conflict is found; false otherwise
+    def conflict?(only: [], show_information: true)
+      textdomain "base"
+      return false if yast_config_file.empty?
+
+      conflicting_attrs = Sysctl::ATTRIBUTES.keys
+      conflicting_attrs &= only unless only.empty?
+      conflicts = {}
+      higher_precedence_files.each do |file|
+        # Checking all "higher" files if their values overrule the current
+        # YAST settings.
+        conflict_values = yast_config_file.conflicts(file) & conflicting_attrs
+        conflicts[file.file_path] = conflict_values unless conflict_values.empty?
+      end
+
+      if !conflicts.empty?
+        log.warn("There are conflicts in sysctl files: #{conflicts}.")
+        log.warn("It could be that #{YAST_CONFIG_PATH} will not be written.")
+        if show_information
+          text = ""
+          text << _("Changed values have conflicts with:<br><br>")
+          conflicts.each do |filename, conflict|
+            text << _("File: %s<br>") % filename
+            text << _("Conflicting entries: %s<br>") % conflict.join(", ")
```

Comment:
It will display the name of the method, but no the name of the key. For instance, it will show `forward_ipv4` instead of `net.ipv4.ip_forward`. You might need to take the renames into account.